components/plurals/src/rules/runtime/resolver.rs

```diff
@@ -0,0 +1,61 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::operands::PluralOperands;
+use crate::rules::runtime::ast;
+
+pub fn test_rule(rule: &ast::Rule, operands: &PluralOperands) -> bool {
+    let mut left = true;
+
+    for relation in rule.0.iter() {
+        let relation = relation.as_relation();
+        if left {
+            if relation.and_or == ast::AndOr::Or {
+                return true;
+            } else {
+                left = test_relation(&relation, operands);
+            }
+        } else if relation.and_or == ast::AndOr::Or {
+            left = test_relation(&relation, operands);
+        }
+    }
+    left
+}
+
+#[inline]
+fn test_relation(relation: &ast::Relation, operands: &PluralOperands) -> bool {
+    let result = if let Some(value) = get_value(relation, operands) {
+        relation.range_list.iter().any(|range| match range {
+            ast::RangeOrValue::Value(v) => u64::from(v) == value,
+            ast::RangeOrValue::Range(min, max) => {
+                value >= u64::from(min) && value <= u64::from(max)
+            }
+        })
+    } else {
+        false
+    };
+    match relation.polarity {
+        ast::Polarity::Negative => !result,
+        ast::Polarity::Positive => result,
+    }
+}
+
+#[inline]
+fn get_value(relation: &ast::Relation, operands: &PluralOperands) -> Option<u64> {
```

Comment:
Question: why `u64` when we are storing all the data as `u32` ?  There's a bit of a conflict here between `u32` and `u64` that we should think about.

Reply:
We are storing `PluralOperands` as a combination of `usize` and `u64` but we are storing `ast::RangeOrValue` as `u32`, so at some point we have to coalesce to compare, and I'm using `u64` for it.

I'm happy to revisit that, but would prefer not to creep the scope of this PR.